🪟 🎨 Aligns Switch design according to the latest Figma updates

YatsukBogdan1 opened the pull request:
## What
Closes #20372

## How
1. Added `size` property for `Switch` component; I omitted native HTMLInput `size` property with TS `Omit` utility type, both in the `Switch` component, and one that is the wrapper for `Switch`; @dizel852 I also did check your `CheckBox` PR, basically we can do the same with `CheckBox` component, by omitting `size` both in `CheckBox` component, and one that is the wrapper for `CheckBox`, there is no other usage of `{...props}`, that can give us the error
2. For `BulkEditPanel` design purposes I also decided to put the `variant` property, for now, it only has `default` and `strong-blue` values, but it can be helpful for us in the future if we will have more colors for `Switch` component
3. I also found a bug on the connection page related to the `Switch` component

https://user-images.githubusercontent.com/25852567/208323499-8b12b4b2-24f3-4ab4-ba57-bb78c1fa8e3d.mov

At first, I thought it was somehow connected to the component itself. As you can see while setting `checked: true`, the component shows the expected animation, but when setting `checked: false` there is no animation. I found out that the issue is related to the `useAsyncFn` `loading` value, which is not correct while updating the connection. You can find it in project by this path
```
airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/EnabledControl.tsx
```
at line 25, not sure why is it working this way, but it seems that any active request should trigger `loading` value to change accordingly

krishnaglick reviewed (approved):
Changes here look good.
The [loading state issue](https://github.com/streamich/react-use/issues?q=is%3Aissue+is%3Aopen+useAsyncFn) comes from react-use.

YatsukBogdan1 commented:
I did couple updates in latest commits:
- added mixed state for `<Switch />` component
- updated failing snapshot tests after adding `area-checked` input property

edmundito commented on airbyte-webapp/src/components/ui/Switch/Switch.tsx:
```diff
@@ -3,29 +3,52 @@ import React from "react";
-  small?: boolean;
+type SwitchSize = "lg" | "sm" | "xs";
+
+type SwitchVariant = "default" | "strong-blue";
```
For props, we agreed to keep using types or unions.
